### PR TITLE
Add viewer_protocol_only_https_compliant query Closes #800. Some typos and errors fixes

### LINKED ISSUE #800: Add only_https_protocol query for AWS CloudFormation
**Platform**
AWS CloudFormation

**Provider**
AWS

**Description**
Check if the Viewer Protocol is only compliant with the HTTPS Protocol

### PR DESCRIPTION
**Pull request by NunoAraujoCX**
Closes #800 

Checks if Viewer Protocol is only compliant with the HTTPS Protocol.
Corrected some typos and errors.

### CODE REVIEW

File: assets/queries/cloudFormation/viewer_protocol_only_https_compliant/metadata.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "viewer_protocol_only_https_compliant",
+  "queryName": "Viewer Protocol Only HTTPS Compliant",
```
Comment (jorge-cruz) [inline assets/queries/cloudFormation/viewer_protocol_only_https_compliant/metadata.json:3]: The name should reflect the vulnerability. A more suitable name would be: "Viewer Protocol Policy Allows HTTP Protocol"

File: assets/queries/cloudFormation/viewer_protocol_only_https_compliant/query.rego [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                "searchKey": 	        sprintf("Resources.%s.Properties.DistributionConfig.DefaultCacheBehavior.ViewerProtocolPolicy", [name]),
+                "issueType":		      "IncorrectValue",
+                "keyExpectedValue":   sprintf("Resources.%s.Properties.DistributionConfig.DefaultCacheBehavior.ViewerProtocolPolicy is 'https-only'", [name]),
+                "keyActualValue": 	  sprintf("Resources.%s.Properties.DistributionConfig.DefaultCacheBehavior.ViewerProtocolPolicy 'isn't https-only'", [name])
+              }
+}
+
+CxPolicy [ result ] {
+  resources := input.document[i].Resources[name]
+  resources.Properties.DistributionConfig.CacheBehaviors.ViewerProtocolPolicy != "https-only"
```
Comment (jorge-cruz) [inline assets/queries/cloudFormation/viewer_protocol_only_https_compliant/query.rego:19]: Cache Behaviors is a list. Replace with:
`resources.Properties.DistributionConfig.CacheBehaviors[_].ViewerProtocolPolicy != "https-only"`

File: assets/queries/cloudFormation/viewer_protocol_only_https_compliant/query.rego [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+	result := {
+                "documentId": 		    input.document[i].id,
+                "searchKey": 	        sprintf("Resources.%s.Properties.DistributionConfig.CacheBehaviors.ViewerProtocolPolicy", [name]),
```
Comment (jorge-cruz) [inline assets/queries/cloudFormation/viewer_protocol_only_https_compliant/query.rego:23]: Replace with:
` "searchKey": 	        sprintf("Resources.%s.Properties.DistributionConfig.CacheBehaviors", [name]),`

File: assets/queries/cloudFormation/viewer_protocol_only_https_compliant/test/positive.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          Value: example
+  cloudfrontdistribution_2:
+    Type: AWS::CloudFront::Distribution
+    Properties:
+      DistributionConfig:
+        CacheBehaviors:
+          ViewerProtocolPolicy: allow-all
+          LambdaFunctionAssociations:
+            - EventType: viewer-request
+              LambdaFunctionARN: examp
```
Comment (jorge-cruz) [inline assets/queries/cloudFormation/viewer_protocol_only_https_compliant/test/positive.yaml:33]: CacheBehaviors is a list. Replace with:
```
          - ViewerProtocolPolicy: allow-all
            LambdaFunctionAssociations:
              - EventType: viewer-request
                LambdaFunctionARN: examp
```

File: assets/queries/cloudFormation/viewer_protocol_only_https_compliant/test/positive_expected_result.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+	{
+		"queryName": "Viewer Protocol Only HTTPS Compliant",
+		"severity": "HIGH",
+		"line": 13
+	}
+	,
+	{
+		"queryName": "Viewer Protocol Only HTTPS Compliant",
+		"severity": "HIGH",
+		"line": 30
```
Comment (jorge-cruz) [inline assets/queries/cloudFormation/viewer_protocol_only_https_compliant/test/positive_expected_result.json:11]: I think this will change to 29

**Review by jorge-cruz — CHANGES_REQUESTED** [review]

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**